lib/graphql/dataloader/loader.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module GraphQL
+  class Dataloader
+    class Loader
+      class StateError < GraphQL::Error; end
+
+      def self.load(context, key, value)
+        dl = context[:dataloader]
+        loader = dl.loaders[self][key]
+        loader.load(value)
+      end
+
+      def initialize(context, key)
+        @context = context
+        @key = key
```

Comment:
Yes, I suppose it could be more readable like:

```ruby 
class MyLoader < GraphQL::Dataloader::Loader
  def initialize(context, model)
    @model = model 
    super 
  end 

  def perform(ids)
    @model.where(id: ids)
  end
end
```

This was _required_ for graphql-batch (IIRC) because it didn't store state otherwise. It also didn't require the `super` call. I wonder how I can remove that boilerplate in this implementation 🤔